server/apiserver/apiserver.go

```diff
@@ -116,7 +116,7 @@ func (api *ApiServer) CallUpdatePixel(x int, y int, r int, g int, b int, userID
 	fmt.Println("\nWithin UpdatePixel")
 
 	// TODO verify that the user is able to update a pizel with the User Data Team
-	userVerification := true
+	userVerification := api.validateUser(userID)
 	if !userVerification {
 		// User verification failed
 
```

Comment:
I can't highlight it in the PR, but below this you also need to take care of the message that is send back when authentication fails. Currently it sends back "FAILURE. TODO make this properly formatted" which is not proper json and was a placeholder for your team. You will need to make a message type for it in the `server\apiserver\message.go` file on the server side and a matching type ID in `client\src\message.ts` for the client. You should add a handler for it on the client side under `client\src\websocket\actions.ts`  in the `websocket.onmessage` section so that the client team can easily add code to to handle it however they will like.

Reply:
I forgot to mention, but you can make functions to wrap strings or integers you want to send like we did with `func makeStatusMessage(s int) []byte` which takes in a string and returns the byte array of it wrapped in the message format.

Reply:
Addressed in the latest commit.